vlog4j-client/src/test/java/org/vlog4j/client/picocli/SaveModelTest.java

```diff
@@ -53,26 +104,125 @@ public void validate_saveModelFalseNullDir_valid() throws ConfigurationException
+	public void printConfiguration_saveModelTrueDefaultDir_valid() {
 		SaveModel sm = new SaveModel();
 		sm.setSaveModel(true);
-		sm.setOutputModelDirectory("query-results");
-		sm.validate();
+		sm.setOutputModelDirectory(defaultDir);
+		assertEquals(String.format(outputConfigurationBase, true, defaultDir), captureOutputPrintConfiguration(sm));
 	}
 
-	@Test(expected = ConfigurationException.class)
-	public void validate_saveModelTrueEmptyDir_notValid() throws ConfigurationException {
+	@Test
+	public void printConfiguration_saveModelTrueValidDir_valid() {
+		// default configuration
+		SaveModel sm = new SaveModel();
+		sm.setSaveModel(false);
+		sm.setOutputModelDirectory(dir);
+		assertEquals(String.format(outputConfigurationBase, false, dir), captureOutputPrintConfiguration(sm));
+	}
+
+	@Test
+	public void printConfiguration_saveModelTrueEmptyDir_notValid() {
 		SaveModel sm = new SaveModel();
 		sm.setSaveModel(true);
 		sm.setOutputModelDirectory("");
-		sm.validate();
+		assertEquals(String.format(outputConfigurationBase, true, ""), captureOutputPrintConfiguration(sm));
 	}
 
-	@Test(expected = ConfigurationException.class)
-	public void validate_saveModelTrueNullDir_notValid() throws ConfigurationException {
+	@Test
+	public void printConfiguration_saveModelTrueNullDir_notValid() {
 		SaveModel sm = new SaveModel();
 		sm.setSaveModel(true);
 		sm.setOutputModelDirectory(null);
-		sm.validate();
+		assertEquals(String.format(outputConfigurationBase, true, null), captureOutputPrintConfiguration(sm));
+	}
+
+	@Test
+	public void printConfiguration_saveModelFalseDefaultDir_valid() {
+		SaveModel sm = new SaveModel();
+		sm.setSaveModel(false);
+		sm.setOutputModelDirectory(defaultDir);
+		assertEquals(String.format(outputConfigurationBase, false, defaultDir), captureOutputPrintConfiguration(sm));
+	}
+
+	@Test
+	public void printConfiguration_saveModelFalseValidDir_valid() {
+		SaveModel sm = new SaveModel();
+		sm.setSaveModel(false);
+		sm.setOutputModelDirectory(dir);
+		assertEquals(String.format(outputConfigurationBase, false, dir), captureOutputPrintConfiguration(sm));
+	}
+
+	@Test
+	public void printConfiguration_saveModelFalseEmptyDir_valid() {
+		SaveModel sm = new SaveModel();
+		sm.setSaveModel(false);
+		sm.setOutputModelDirectory("");
+		assertEquals(String.format(outputConfigurationBase, false, ""), captureOutputPrintConfiguration(sm));
+	}
+
+	@Test
+	public void printConfiguration_saveModelFalseNullDir_valid() {
+		SaveModel sm = new SaveModel();
+		sm.setSaveModel(false);
+		sm.setOutputModelDirectory(null);
+		assertEquals(String.format(outputConfigurationBase, false, null), captureOutputPrintConfiguration(sm));
+	}
+
+	@Test
+	public void prepare_saveModelTrueValidDir() {
+		SaveModel sm = new SaveModel();
+		sm.setSaveModel(true);
+		sm.setOutputModelDirectory(tempDir);
+		sm.prepare();
+		File f = new File(tempDir);
+		assert (f.exists() && f.isDirectory());
+		f.delete();
```

Comment:
Perhaps Mockito can be used to simulate creating a directory, instead of actually creating one and then deleting it.